### ISSUE TITLE
Generated mode not work, connect ECONNREFUSED

### ISSUE THREAD

**Issue by n4an** [nuxt/framework#8155]
### Environment

nuxt 3

### Reproduction

See previous issue

### Describe the bug

Generate mode not work. With rc11 when i run generate and after start and open site in browser i see list of directories (Index of  public/en) and not html page.

And when I see list of generated pages many pages are not created, not created pages which links is created with v-if from data that is loaded with fetch called from pinia.

```
Nuxi 3.0.0-rc.11                                                                                                                                                                     10:17:18
ℹ Node.js version: 16.17.1                                                                                                                                                           10:17:18
ℹ Preset: nitro-prerender                                                                                                                                                            10:17:18
ℹ Working dir: .output                                                                                                                                                               10:17:18
ℹ Starting preview command: npx serve -s ./public     
```

**Comment by n4an** [comment]
So, I not understand if generated mode work

**Comment by n4an** [comment]
rc12 also not work show list of catalog. In past it show some page without (repeated items with v-if form data from pinia) content at leas.

**Comment by manniL** [comment]
Your content is only fetched client-side:

```js
onMounted (async () => {
  await listStore.initiateList({
    element: 'test',
  })
  update()
})
```

**Comment by n4an** [comment]
@manniL but it not show html any more

**Comment by manniL** [comment]
@n4an It is really hard to follow the conversation 🙈 
* Please try to write all info in one comment instead of creating multiple ones below each other
* This is likely not a Nuxt issue but an issue with your app. Feel free to create a discussion instead ☺️ 

Using `onMounted` will render data on client-side only. You probably want to use the call direct in `<script setup>`.

**Comment by danielroe** [comment]
This is the underlying issue: https://github.com/nuxt/framework/issues/6226.

**Comment by n4an** [comment]
> This is the underlying issue: #6226.

@danielroe I remove onmout hook and code is in setup now and pages not generated (i will try more times)

**Comment by danielroe** [comment]
Nitro prerender pages with query parameters in them.

**Comment by n4an** [comment]
> Nitro prerender pages with query parameters in them.

In Nuxt 2 that code works well.

**Comment by n4an** [comment]
> Nitro prerender pages with query parameters in them.

Not understand, but without query parameters are the same 

Failed to load resource: the server responded with a status of 404 (Not Found)
image.png:1          
Failed to load resource: the server responded with a status of 404 (Not Found) 
api/test

**Comment by danielroe** [comment]
If you would provide a reproduction I will happily look into it.

**Comment by n4an** [comment]
> Nitro prerender pages with query parameters in them

It was, but I updated it (remove unmounted, and add new page and link to it).
Here is it updated https://github.com/n4an/nuxt-error-7

In dev mode it worked with onMounted hook, but without not work.
The store action is called, the data is fetched but that data is not used to render list :(.
In nuxt 2 code was in created hook not in onMounted and works well. 
But direct in setup not.

**Comment by n4an** [comment]
@danielroe @manniL something not work

**Comment by n4an** [comment]
May be to convert

```
await useAsyncData(payload.element, () => 
      $http.$get({
        url: `/api/list`
      })
    )
      .then((res: any) => {
        data.value[payload.element]['items'] = res.data.value.items
      })
      .catch((error: any) => {
        // console.log(error)
      })
```

 to 


```
const { data, pending, error, refresh } = await useAsyncData(
  'mountains',
  () => $fetch('https://api.nuxtjs.dev/mountains')
)
```

**Comment by danielroe** [comment]
indeed, `nuxt/http` is not compatible with nuxt 3.

**Comment by n4an** [comment]
I understand, is need to add await to useAsyncData and in other.

**Comment by n4an** [comment]
sorry, problem is with async/await and useAsyncData

**Comment by n4an** [comment]
@danielroe but it also not generate all pages (test page) :(

**Comment by n4an** [comment]
@danielroe now with $fetch work well

```
await useAsyncData(payload.element, () => 
      $fetch(`/api/list`, {
        method: 'get',
      })
    )
      .then((res: any) => {console.log(res)
        data.value[payload.element]['items'] = res.data.value.items
      })
      .catch((error: any) => {
        // console.log(error)
      })
      ```

**Comment by danielroe** [comment]
please don't assign to `data.value` that way. With `useAsyncData` you need to return the value from the fetcher.

Also, may I suggest creating a discussion or joining Discord for support?

**Comment by n4an** [comment]
but why?

**Comment by n4an** [comment]
Uraaaaaaaaaa it works with my module http and auth. But not work with https://github.com/Teranode/nuxt-module-alternatives/tree/master/%40nuxtjs-alt

Teranode make the shit!!! 
Go to say to he good news :)

**Comment by n4an** [comment]
Why it work in dev and not work in ssg mode?

**Comment by n4an** [comment]
It not work because of

```
ℹ Building server...                                                                                                                                                                 19:24:01
✔ Server built in 3080ms                                                                                                                                                             19:24:04
✔ Generated public .output/public                                                                                                                                              nitro 19:24:04
ℹ Initializing prerenderer                                                                                                                                                     nitro 19:24:04
ℹ Prerendering 3 initial routes with crawler                                                                                                                                   nitro 19:24:07
{                                                                                                                                                                                    19:24:08
  data: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: FetchError: request to http://localhost:3000/api/list failed, reason: connect ECONNREFUSED 127.0.0.1:3000 ()

```

**Comment by n4an** [comment]
> Please provide a reproduction.

@danielroe here https://github.com/n4an/nuxt-error-7

**Comment by n4an** [comment]
?

**Comment by danielroe** [comment]
What was the issue?

**Comment by n4an** [comment]
@teranode say that error was eliminated. but new one https://github.com/Teranode/nuxt-module-alternatives/issues/101

**Comment by n4an** [comment]
[Vue warn]: onServerPrefetch is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.
FetchError: 500  (/api/test?page=1&limit=0)

**Comment by n4an** [comment]
Such error on some paths
 {                                                                                                                                                                              
  data: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: null,
    _value: null
  },
  pending: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: false,
    _value: false
  },
  error: RefImpl {
    __v_isShallow: false,
    dep: undefined,
    __v_isRef: true,
    _rawValue: FetchError: 500  (/api/test/index)

**Comment by n4an** [comment]
and on some page with disable js all content is ok, but when it is on content disappear. I do something wrong again

**Comment by n4an** [comment]
I open page in browser /en/test/2/3/8
 in console - FetchError: 404 Not Found (http://localhost:3000/api/test/8) 
 but file exists en/test/2/3/8/_payload.js 
  
@danielroe  Generate work with pages like /test/[...slug].vue (en/test/2/3/8/ or en/test/2/3/7/8/)?

**Comment by n4an** [comment]
I open page in browser /en/test/2/3/8
in console - FetchError: 404 Not Found (http://localhost:3000/api/test/8)
but file exists en/test/2/3/8/_payload.js

@danielroe Generate work with pages like /test/[...slug].vue (en/test/2/3/8/ or en/test/2/3/7/8/)?

**Comment by n4an** [comment]
@danielroe don't know what i do but now when i navigate through pages i have 2 and more pages in one

**Comment by n4an** [comment]
http://localhost:3000/_payload.js net::ERR_ABORTED 404 (Not Found)

**Comment by danielroe** [comment]
That issue should be resolved with https://github.com/nuxt/framework/pull/7805.

**Comment by n4an** [comment]
Crawler try to generate /en/index and not /en

/__nuxt_error?url=/en/index&statusCode=404&statusMessage=Page+not+found:+/en/index&message=Page+not+found:+/en/index&stack&data

**Comment by n4an** [comment]
When navigate to http://localhost:3000/ru/test/2/3/5?page=2
> GET http://localhost:3000/api/test/2/3/5 404 (Not Found)

**Comment by n4an** [comment]
and that cause to me troubles

runtime-core.esm-bundler.js:22 Uncaught (in promise) TypeError: Cannot convert a Symbol value to a string
    at Array.join (<anonymous>)
    at warn2 (runtime-core.esm-bundler.js:22:24)
    at handleMismatch (runtime-core.esm-bundler.js:4895:13)
    at onMismatch (runtime-core.esm-bundler.js:4618:34)
    at hydrateNode (runtime-core.esm-bundler.js:4654:32)
    at hydrateSubTree (runtime-core.esm-bundler.js:5594:25)
    at ReactiveEffect.componentUpdateFn [as fn] (runtime-core.esm-bundler.js:5608:25)
    at ReactiveEffect.run (reactivity.esm-bundler.js:187:25)
    at instance.update (runtime-core.esm-bundler.js:5729:56)
    at setupRenderEffect (runtime-core.esm-bundler.js:5743:9)

**Comment by n4an** [comment]
Not work on pagination http://localhost:3000/ru/test/2/3/5?page=4  (api /test/[...slug].get.ts)
```
entry.31a00ebd.js:1  GET http://localhost:3000/api/test/2/3/5?page=4&limit=0 404 (Not Found)
entry.31a00ebd.js:1  GET http://localhost:3000/_payload.js net::ERR_ABORTED 404 (Not Found)
entry.31a00ebd.js:1 [nuxt] Cannot load payload  /_payload.js TypeError: Failed to fetch dynamically imported module: http://localhost:3000/_payload.js
```

and images 
`GET http://localhost:3000/_ipx/_/img/image.png 404 (Not Found)`
